index.js

```diff
@@ -1,10 +1,18 @@
 'use strict';
 
+// Polyfill Promise if running on Node < 0.11
+// Note:  Must modify global Promise for buffered-spawn
+if ('undefined' === typeof Promise) {
+  global.Promise = require('bluebird').Promise;
+}
+
 var spawn = require('cross-spawn')
   , which = require('which')
   , path = require('path')
   , util = require('util')
-  , tty = require('tty');
+  , tty = require('tty')
+  , bufferedSpawn = require('buffered-spawn')
```

Comment:
I tend to think of synchronous execution as an anti-pattern since it prevents the addition of concurrency.  But if you feel strongly about it, I can rework the commit to use only synchronous methods.